// skystream-extractors/dist/extractors/mix_drop.js
import { ExtractorApi } from './core/extractor_api.js';
import { Qualities } from './core/qualities.js';
import { JsUnpacker } from './utils/js_unpacker.js';
export class MixDrop extends ExtractorApi {
    constructor() {
        super(...arguments);
        this.name = 'MixDrop';
        this.mainUrl = 'https://mixdrop.co';
        this.requiresReferer = false;
    }
    async getUrl(url, referer) {
        const res = await http_get(url, referer ? { Referer: referer } : {});
        if (res.status !== 200)
            return [];
        const unpacked = JsUnpacker.unpack(res.body);
        const match = /wurl="(https?:[^"]+)"/.exec(unpacked);
        if (!match)
            return [];
        let finalUrl = match[1];
        if (finalUrl.startsWith('//')) {
            finalUrl = 'https:' + finalUrl;
        }
        return [{
                name: 'MixDrop',
                source: 'MixDrop',
                url: finalUrl,
                quality: Qualities.Unknown,
                type: 'video',
                headers: { Referer: this.mainUrl }
            }];
    }
}
