// skystream-extractors/dist/extractors/vid_hide_pro.js
import { ExtractorApi } from './core/extractor_api.js';
import { Qualities } from './core/qualities.js';
import { JsUnpacker } from './utils/js_unpacker.js';
export class VidHidePro extends ExtractorApi {
    constructor() {
        super(...arguments);
        this.name = 'VidHidePro';
        this.mainUrl = 'https://vidhidepro.com';
        this.requiresReferer = false;
    }
    async getUrl(url, referer) {
        const res = await http_get(url);
        if (res.status !== 200)
            return [];
        let body = res.body;
        if (body.includes('p,a,c,k,e,d')) {
            body = JsUnpacker.unpack(body);
        }
        const fileMatch = /file:\s*["']([^"']+\.m3u8[^"']*)["']/.exec(body) || /sources:\s*\[{file:\s*["']([^"']+)["']/.exec(body);
        if (!fileMatch)
            return [];
        return [{
                name: this.name,
                source: this.name,
                url: fileMatch[1],
                quality: Qualities.Unknown,
                type: 'm3u8',
                headers: { Referer: this.mainUrl }
            }];
    }
}
