// skystream-extractors/dist/extractors/voe.js
import { ExtractorApi } from './core/extractor_api.js';
import { Qualities } from './core/qualities.js';
export class Voe extends ExtractorApi {
    constructor() {
        super(...arguments);
        this.name = 'Voe';
        this.mainUrl = 'https://voe.sx';
        this.requiresReferer = false;
    }
    async getUrl(url, referer) {
        const res = await http_get(url);
        if (res.status !== 200)
            return [];
        // Voe usually stores the master m3u8 in a script variable, often something like `var sources = { hls: "..." }`
        // or inside a specific script tag.
        const hlsMatch = /'hls':\s*'([^']+)'/.exec(res.body) || /"hls":\s*"([^"]+)"/.exec(res.body);
        const videoUrl = hlsMatch ? hlsMatch[1] : null;
        if (!videoUrl)
            return [];
        return [{
                name: this.name,
                source: this.name,
                url: videoUrl,
                quality: Qualities.Unknown,
                type: 'm3u8',
                headers: { Referer: this.mainUrl }
            }];
    }
}
